odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -1280,6 +1280,134 @@ def test_27_company_dependent(self):
         self.assertEqual(attribute_record.company.foo, 'DEF')
         self.assertEqual(attribute_record.bar, 'DEFDEF')
 
+    def test_28_company_dependent_search(self):
+        """ Test the search on company-dependent fields in all corner cases.
+            This assumes that get_multi/set_multi are correct, and that
+            filtered_domain() correctly mimics the search.
+        """
+        Property = self.env['ir.property']
+        Model = self.env['test_new_api.company']
+        date1, date2 = date(2021, 11, 22), date(2021, 11, 23)
+        moment1, moment2 = datetime(2021, 11, 22), datetime(2021, 11, 23)
+        tag1, tag2 = self.env['test_new_api.multi.tag'].create([{'name': 'one'}, {'name': 'two'}])
+
+        # description of cases: (field_name, truthy_record_values, {operation: test_values})
+        CASES = [
+            # boolean fields
+            ('truth', [True, True], {
+                '=': (True, False),
+                '!=': (True, False),
```

Comment:
Why is this even a hashmap? That only seems more confusing.

Reply:
IMO, it is more readable like this than : 
```python
('truth', [True, True], (
    ('=', (True, False)),
    ('!=', (True, False)),
)),
```